common/dpcpp/slab_hash.hpp

```diff
@@ -0,0 +1,182 @@
+#pragma once
+
+#include <CL/sycl.hpp>
+#include <algorithm>
+#include "dpcpp_common.hpp"
+
+#define SUBGROUP_SIZE 16
+#define CONST 64
```

Comment:
Very confusing naming, try renaming. Also, why not using constexprs instead of defines ?

Reply:
I have no reason to use defines, but why is it better to use constexpr in this cases?

Reply:
They follow c++ semantics, ie. respect scope. Since you don't use anything like #ifdef, there's no much reason to drop constexpr advantages.

Reply:
ok, thanks!